catalog-rest-service/src/main/java/org/openmetadata/catalog/security/JwtFilter.java

```diff
@@ -80,24 +82,27 @@ public void filter(ContainerRequestContext requestContext) {
     } catch (RuntimeException runtimeException) {
       throw new AuthenticationException("Invalid token");
     }
-    String authorizedEmail;
-    if (jwt.getClaims().get(jwtEmail) != null) {
-      authorizedEmail = jwt.getClaim(jwtEmail).as(TextNode.class).asText();
-    } else if (jwt.getClaims().get("email") != null) {
-      authorizedEmail = jwt.getClaim("email").as(TextNode.class).asText();
-    } else if (jwt.getClaims().get("preferred_username") != null) {
-      authorizedEmail = jwt.getClaim("preferred_username").as(TextNode.class).asText();
-    } else if (jwt.getClaim("sub") != null) {
-      authorizedEmail = jwt.getClaim("sub").as(TextNode.class).asText();
-    } else {
-      throw new AuthenticationException("Invalid JWT token, \"email\" or \"subject\" not present.");
-    }
-    String userName;
-    if (authorizedEmail.contains("@")) {
-      userName = authorizedEmail.split("@")[0];
-    } else {
-      userName = authorizedEmail;
-    }
+
+    // Get username from JWT token
+    String userName =
+        jwtPrincipalClaims.stream()
+            .filter(jwt.getClaims()::containsKey)
+            .findFirst()
+            .map(jwt::getClaim)
+            .map(claim -> claim.as(TextNode.class).asText())
+            .map(
+                authorizedEmail -> {
```

Comment:
Can we rename this variable to something generic since it can be an email or username?